pandas/io/common.py

```diff
@@ -322,18 +309,19 @@ def get_filepath_or_buffer(
             "storage_options passed with file object or non-fsspec file path"
         )
 
-    if isinstance(filepath_or_buffer, (str, bytes, mmap.mmap)):
-        return IOargs(
-            filepath_or_buffer=_expand_user(filepath_or_buffer),
-            encoding=encoding,
-            compression=compression,
-            should_close=False,
-            mode=mode,
-        )
+    if is_file_like(filepath_or_buffer) or isinstance(filepath_or_buffer, mmap.mmap):
```

Comment:
can we change the definition of is_file_like to *include* mmamp.mmap (e.g. does this break anything generally)?

Reply:
since mmap.mmap objects have read/write methods and are iterables, it seems very much in line with the docstring in `is_file_like`, so I think this is an excellent idea.